subsys/bluetooth/services/cgms/cgms_racp.c

```diff
@@ -0,0 +1,489 @@
+	rsp[1] = RACP_OPERATOR_NULL;
+	rsp[2] = count & 0xFF;
+	rsp[3] = (count >> 8) & 0xFF;
+
+	rc = cgms_racp_send_response(peer, rsp, sizeof(rsp));
+	return rc;
+}
+
+static int report_num_recs_greater_or_equal_handler(struct bt_conn *peer,
+				uint8_t *req_operand, uint16_t len)
+{
+	int rc;
+	uint16_t time_offset_limit;
+	uint16_t count = 0;
+	uint8_t rsp[4];
+	struct cgms_meas_db_entry *entry;
+	sys_snode_t *record_node;
+
+	if (req_operand[0] != RACP_OPERAND_FILTER_TYPE_TIME_OFFSET) {
+		rc = generic_handler(peer,
+			RACP_OPCODE_REPORT_NUM_RECS, RACP_RESPONSE_INVALID_OPERATOR);
+		return rc;
+	}
+
+	if (len < 3) {
+		rc = generic_handler(peer,
+			RACP_OPCODE_REPORT_RECS, RACP_RESPONSE_INVALID_OPERAND);
+		return rc;
+	}
+
+	time_offset_limit = req_operand[1] | req_operand[2] << 8;
+
+	if (sys_slist_is_empty(&database)) {
+		count = 0;
+	} else {
+		SYS_SLIST_FOR_EACH_NODE(&database, record_node) {
+			entry = CONTAINER_OF(record_node, struct cgms_meas_db_entry, node);
+			if (entry->meas.time_offset >= time_offset_limit) {
+				count++;
+			}
+		}
+	}
+
+	rsp[0] = RACP_OPCODE_NUM_RECS_RESPONSE;
+	rsp[1] = RACP_OPERATOR_NULL;
+	rsp[2] = count & 0xFF;
+	rsp[3] = (count>>8) & 0xFF;
+
+	rc = cgms_racp_send_response(peer, rsp, sizeof(rsp));
+	return rc;
+}
+
+static int report_num_recs_handler(struct bt_conn *peer,
+			uint8_t *req_opreator_and_operand, uint16_t len)
+{
+	int rc;
+
+	if (len < 1) {
+		rc = generic_handler(peer, RACP_OPCODE_REPORT_NUM_RECS,
+			RACP_RESPONSE_INVALID_OPERATOR);
+		return rc;
+	}
+	if (req_opreator_and_operand[0] == RACP_OPERATOR_ALL) {
+		rc = report_num_recs_all_handler(peer,
+					len - 1 > 0 ? &req_opreator_and_operand[1] : NULL,
+					len - 1 > 0 ? len - 1 : 0);
+		return rc;
+	}
+	if (req_opreator_and_operand[0] == RACP_OPERATOR_GREATER_OR_EQUAL) {
+		rc = report_num_recs_greater_or_equal_handler(peer,
+					len - 1 > 0 ? &req_opreator_and_operand[1] : NULL,
+					len - 1 > 0 ? len - 1 : 0);
+		return rc;
+	}
+
+	rc = generic_handler(peer, RACP_OPCODE_REPORT_NUM_RECS, RACP_RESPONSE_OPERATOR_UNSUPPORTED);
+	return rc;
+}
+
+static void racp_task_handler(struct k_work *work_item)
+{
+	int rc;
+	struct racp_task *task;
+
+	task = CONTAINER_OF(work_item, struct racp_task, item);
+
+	if (k_mutex_lock(&lock, K_NO_WAIT) == 0) {
+		switch (task->req_data[0]) {
+		case RACP_OPCODE_REPORT_RECS:
+			rc = report_recs_handler(task->peer,
+				&task->req_data[1], task->req_len - 1);
+			break;
+		case RACP_OPCODE_REPORT_NUM_RECS:
+			rc = report_num_recs_handler(task->peer,
+				&task->req_data[1], task->req_len - 1);
+			break;
+		default:
+			rc = generic_handler(task->peer, task->req_data[0],
+					RACP_RESPONSE_OPCODE_UNSUPPORTED);
+			break;
+		}
+		k_mutex_unlock(&lock);
+	} else {
+		rc = -EBUSY;
+	}
+}
+
+int cgms_racp_recv_request(struct bt_conn *peer, const uint8_t *req_data, uint16_t req_len)
+{
+	int rc;
+
+	/* Early check of the data length. */
+	if (req_data == NULL || req_len == 0) {
```

Comment:
I think we use the following syntax in NCS for NULL checks (you can align it in other places)

```suggestion
	if (!req_data || req_len == 0) {
```